### PR TITLE
chore(threads): Analytics

### PR DESCRIPTION
**Pull request by ZaymonFC**
- Resolves: https://github.com/dxos/devrel/issues/97

### CODE REVIEW

File: packages/apps/plugins/plugin-thread/src/hooks/analytics.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { useOnTransition } from '@dxos/react-ui';
+
+export const useAnalyticsCallback = (space: string | undefined, name: string, meta?: any) => {
+  const dispatch = useIntentDispatcher();
+
+  return useCallback(
+    (dynamicMeta?: any) => {
+      void dispatch({
+        action: ObservabilityAction.SEND_EVENT,
+        data: { name, ...meta, ...dynamicMeta, space },
```
Comment (wittjosiah) [inline packages/apps/plugins/plugin-thread/src/hooks/analytics.ts:20]: also all of the non-event name properties need to be inside `properties` field

File: packages/apps/plugins/plugin-thread/src/hooks/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    [dispatch, name, meta, space],
+  );
+};
+
+export const useOnEditAnalytics = (message: ReactiveObject<any>, editing: boolean) => {
+  const space = getSpace(message);
+
+  const onEditAnalytics = useAnalyticsCallback(space?.id, 'threads.comment-edited', {
+    messageId: message.id,
+    messageLength: message.text.length,
```
Comment (wittjosiah) [inline packages/apps/plugins/plugin-thread/src/hooks/analytics.ts:32]: will this capture the initial length or the edited length? probably the initial?

Comment (ZaymonFC) [reply packages/apps/plugins/plugin-thread/src/hooks/analytics.ts:32]: Pretty sure it's mutating the doc on keypress like our other editors do so it should be relatively up to date. Clicking done-editing is a bit of a formality right.

File: packages/apps/plugins/plugin-thread/src/hooks/analytics.ts (hunk 1 above)
Comment (ZaymonFC) [reply packages/apps/plugins/plugin-thread/src/hooks/analytics.ts]: Ahh. Sweet.

**Pull request merged**